vagrant/plugins/vagrant-hypconfigmgmt/spec/unit/command/ensure_default_domain_configured_spec.rb

```diff
@@ -0,0 +1,57 @@
+# -*- encoding: utf-8 -*-
+# vim: set fileencoding=utf-8
+
+require 'spec_helper'
+require "vagrant-hypconfigmgmt/command"
+
+describe VagrantHypconfigmgmt::Command do
+  # create a fake app and env to pass into the VagrantHypconfigmgmt::Command constructor
+  let(:app) { }
+  let(:env) { { :ui => ui } }
+
+  # pretend env contains the Vagrant ui element
+  let(:ui) do
```

Comment:
this is a mocked UI object, so we can test that the proper methods have been called. for a simpler example of how Vagrant plugins are structured see this [small inline one that we also use](https://github.com/ByteInternet/hypernode-vagrant/blob/master/vagrant/plugins/inline/ensure-varnish.rb)